<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <link href='https://fonts.googleapis.com/css?family=Architects+Daughter' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print">

    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <title>Awesome-single-cell by seandavi</title>
  </head>

  <body>
    <header>
      <div class="inner">
        <h1>Awesome-single-cell</h1>
        <h2>List of software packages for single-cell data analysis, including RNA-seq, ATAC-seq, etc.</h2>
        <a href="https://github.com/seandavi/awesome-single-cell" class="button"><small>View project on</small> GitHub</a>
      </div>
    </header>

    <div id="content-wrapper">
      <div class="inner clearfix">
        <section id="main-content">
          <h1>
<a id="awesome-single-cell" class="anchor" href="#awesome-single-cell" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>awesome-single-cell</h1>

<p>List of software packages (and the people developing these methods) for single-cell data analysis, including RNA-seq, ATAC-seq, etc. <a href="https://github.com/seandavi/awesome-single-cell/blob/master/CONTRIBUTE.md">Contributions welcome</a>...</p>

<h2>
<a id="software-packages" class="anchor" href="#software-packages" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Software packages</h2>

<h3>
<a id="rna-seq" class="anchor" href="#rna-seq" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>RNA-seq</h3>

<ul>
<li>
<a href="https://github.com/yeolab/anchor">anchor</a> - [Python] - <g-emoji alias="anchor" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/2693.png">‚öì</g-emoji> Find bimodal, unimodal, and multimodal features in your data</li>
<li>
<a href="https://github.com/linnarsson-lab/BackSPIN">BackSPIN</a> - [Python] - Biclustering algorithm developed taking into account intrinsic features of single-cell RNA-seq experiments.</li>
<li>
<a href="https://github.com/catavallejos/BASiCS">BASiCs</a> - [R] - Bayesian Analysis of single-cell RNA-seq data. Estimates cell-specific normalization constants. Technical variability is quantified based on spike-in genes. The total variability of the expression counts is decomposed into technical and biological components.</li>
<li>
<a href="https://github.com/yeolab/bonvoyage">bonvoyage</a> - [Python] - <g-emoji alias="triangular_ruler" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d0.png">üìê</g-emoji> Transform percentage-based units into a 2d space to evaluate changes in distribution with both magnitude and direction.</li>
<li>
<a href="https://github.com/nghiavtr/BPSC">BPSC</a> - [R] - Beta-Poisson model for single-cell RNA-seq data analyses</li>
<li>
<a href="https://github.com/teichlab/cellity">Cellity</a> - [R] - Classification of low quality cells in scRNA-seq data using R</li>
<li>
<a href="https://www.bioconductor.org/packages/3.3/bioc/html/cellTree.html">cellTree</a> - [R] - Cell population analysis and visualization from single cell RNA-seq data using a Latent Dirichlet Allocation model.</li>
<li>
<a href="https://github.com/epurdom/clusterExperiment">clusterExperiment</a> - [R] - Functions for running and comparing many different clusterings of single-cell sequencing data. Meant to work with SCONE and slingshot.</li>
<li>
<a href="https://github.com/GGiecold/ECLAIR">ECLAIR</a> - [python] - ECLAIR stands for Ensemble Clustering for Lineage Analysis, Inference and Robustness. Robust and scalable inference of cell lineages from gene expression data.</li>
<li>
<a href="https://github.com/VCCRI/Falco/">Falco</a> - [AWS cloud] - <a href="http://www.biorxiv.org/content/early/2016/07/15/064006.abstract">Falco: A quick and flexible single-cell RNA-seq processing framework on the cloud</a>.</li>
<li>
<a href="https://github.com/yeolab/flotilla">flotilla</a> - [Python] Reproducible machine learning analysis of gene expression and alternative splicing data</li>
<li>
<a href="https://github.com/joeburns06/hocuspocus">HocusPocus</a> - [R] - Basic PCA-based workflow for analysis and plotting of single cell RNA-seq data.</li>
<li>
<a href="https://github.com/RGLab/MAST">MAST</a> - [R] - Model-based Analysis of Single-cell Transcriptomics
(MAST) fits a two-part, generalized linear models that are specially adapted for bimodal and/or zero-inflated single cell gene expression data.</li>
<li>
<a href="http://cole-trapnell-lab.github.io/monocle-release/">Monocle</a> - [R] - Differential expression and time-series analysis for single-cell RNA-Seq.</li>
<li>
<a href="https://github.com/lengning/OEFinder">OEFinder</a> - [R] - Identify ordering effect genes in single cell RNA-seq data. OEFinder shiny impelemention depends on packages shiny, shinyFiles, gdata, and EBSeq.</li>
<li>
<a href="https://github.com/kieranrcampbell/ouija">Ouija</a> - [R] - Incorporate prior information into single-cell trajectory (pseudotime) analyses using Bayesian nonlinear factor analysis.</li>
<li>
<a href="https://github.com/YeoLab/outrigger">outrigger</a> - [Python] - Outrigger is a program to calculate alternative splicing scores of RNA-Seq data based on junction reads and a <em>de novo</em>, custom annotation created with a graph database, especially made for single-cell analyses.</li>
<li>
<a href="https://github.com/hemberg-lab/sc3">SC3</a> - [R] - SC3 is an interactive tool for the unsupervised clustering of cells from single cell RNA-Seq experiments.</li>
<li>
<a href="bioconductor.org/packages/scater">scater</a> - [R] - Scater places an emphasis on tools for quality control, visualisation and pre-processing of data before further downstream analysis, filling a useful niche between raw RNA-sequencing count or transcripts-per-million data and more focused downstream modelling tools such as monocle, scLVM, SCDE, edgeR, limma and so on.</li>
<li>
<a href="https://github.com/kdkorthauer/scDD">scDD</a> - [R] - scDD (Single-Cell Differential Distributions) is a framework to identify genes with different expression patterns between biological groups of interest.  In addition to traditional differential expression, it can detect differences that are more complex and subtle than a mean shift.</li>
<li>
<a href="https://github.com/hms-dbmi/scde">SCDE</a> - [R] - Differential expression using error models and overdispersion-based identification of important gene sets.</li>
<li>
<a href="https://github.com/diazlab/SCell">SCell</a> - [matlab] - SCell is an integrated software tool for quality filtering, normalization, feature selection, iterative dimensionality reduction, clustering and the estimation of gene-expression gradients from large ensembles of single-cell RNA-seq datasets. SCell is open source, and implemented with an intuitive graphical interface.</li>
<li>
<a href="https://github.com/PMBio/scLVM">scLVM</a> - [R] - scLVM is a modelling framework for single-cell RNA-seq data that can be used to dissect the observed heterogeneity into different sources, thereby allowing for the correction of confounding sources of variation.  scLVM was primarily designed to account for cell-cycle induced variations in single-cell RNA-seq data where cell cycle is the primary soure of variability. </li>
<li>
<a href="https://github.com/YosefLab/scone">SCONE</a> - [R] - SCONE (Single-Cell Overview of Normalized Expression), a package for single-cell RNA-seq data quality control (QC) and normalization. This data-driven framework uses summaries of expression data to assess the efficacy of normalization workflows.</li>
<li>
<a href="https://github.com/hmatsu1226/SCOUP">SCOUP</a> - [C++] - Uses probabilistic model based on the Ornstein-Uhlenbeck process to analyze single-cell expression data during differentiation. </li>
<li>
<a href="http://bioconductor.org/packages/scran">scran</a> - [R] - This package implements a variety of low-level analyses of single-cell RNA-seq data. Methods are provided for normalization of cell-specific biases, pool-based norms to estimate size factors, assignment of cell cycle phase, and detection of highly variable and significantly correlated genes.</li>
<li>
<a href="https://github.com/zji90/SCRAT">SCRAT</a> - [R] - SCRAT provides essential tools for users to read in single-cell regolome data (ChIP-seq, ATAC-seq, DNase-seq) and summarize into different types of features. It also allows users to visualize the features, cluster samples and identify key features.</li>
<li>
<a href="https://github.com/zji90/SEPA">SEPA</a> - [R] - SEPA provides convenient functions for users to assign genes into different gene expression patterns such as constant, monotone increasing and increasing then decreasing. SEPA then performs GO enrichment analysis to analysis the functional roles of genes with same or similar patterns.</li>
<li>
<a href="https://github.com/ElementoLab/scTCRseq">scTCRseq</a> - [python] - Map T-cell receptor (TCR) repertoires from single cell RNAseq.</li>
<li>
<a href="http://www.satijalab.org/seurat.html">Seurat</a> - [R] - It contains easy-to-use implementations of commonly used analytical techniques, including the identification of highly variable genes, dimensionality reduction (PCA, ICA, t-SNE), standard unsupervised clustering algorithms (density clustering, hierarchical clustering, k-means), and the discovery of differentially expressed genes and markers.</li>
<li>
<a href="http://bioconductor.org/packages/sincell">sincell</a> - [R] - Existing computational approaches for the assessment of cell-state hierarchies from single-cell data might be formalized under a general workflow composed of i) a metric to assess cell-to-cell similarities (combined or not with a dimensionality reduction step), and ii) a graph-building algorithm (optionally making use of a cells-clustering step). Sincell R package implements a methodological toolbox allowing flexible workflows under such framework.</li>
<li>
<a href="https://research.cchmc.org/pbge/sincera.html">sincera</a> - [R] - R-based pipeline for single-cell analysis including clustering and visualization.</li>
<li>
<a href="http://www.morgridge.net/SinQC.html">SinQC</a> - [R] - A Method and Tool to Control Single-cell RNA-seq Data Quality.</li>
<li>
<a href="https://github.com/jw156605/SLICER">SLICER</a> - [R] - Selective Locally linear Inference of Cellular Expression Relationships (SLICER) algorithm for inferring cell trajectories.</li>
<li>
<a href="https://github.com/kstreet13/slingshot">slingshot</a> - [R] - Functions for identifying and characterizing continuous developmental trajectories in single-cell sequencing data.</li>
<li>
<a href="http://www.nature.com/nprot/journal/v11/n7/full/nprot.2016.066.html">SPADE</a> - [R] - Visualization and cellular hierarchy inference of single-cell data using SPADE.</li>
<li>
<a href="http://github.com/kieranrcampbell/switchde">switchde</a> - [R] - Differential expression analysis across pseudotime. Identify genes that exhibit switch-like up or down regulation along single-cell trajectories along with where in the trajectory the regulation occurs.</li>
<li>
<a href="http://github.com/teichlab/tracer">TraCeR</a> - [python] - Reconstruction of T-Cell receptor sequences from single-cell RNA-seq data.</li>
<li>
<a href="https://github.com/zji90/TSCAN">TSCAN</a> - [R] - Pseudo-time reconstruction and evaluation in single-cell RNA-seq analysis.</li>
</ul>

<h3>
<a id="copy-number-analysis" class="anchor" href="#copy-number-analysis" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Copy number analysis</h3>

<ul>
<li>
<a href="https://github.com/robertaboukhalil/ginkgo">Gingko</a> - [R, C] - Gingko is a cloud-based web app single-cell copy-number variation analysis tool.</li>
</ul>

<h2>
<a id="tutorials-and-workflows" class="anchor" href="#tutorials-and-workflows" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Tutorials and workflows</h2>

<ul>
<li>
<a href="http://bioconductor.org/help/workflows/simpleSingleCell/">Aaron Lun's Single Cell workflow on Bioconductor</a> - [R] - This article describes a computational workflow for basic analysis of scRNA-seq data using software packages from the open-source Bioconductor project.</li>
<li>
<a href="https://github.com/drisso/bioc2016singlecell">Bioconductor2016 Single-cell-RNA-sequencing workshop by Sandrine Dudoit lab</a> - [R] - SCONE, clusterExperiment, and slingshot tutorial.</li>
<li>
<a href="http://www.biomedcentral.com/collections/singlecellomics">BiomedCentral Single Cell Omics collectin</a> - collection of papers describing techniques for single-cell analysis and protocols.</li>
<li>
<a href="https://github.com/YeoLab/single-cell-bioinformatics/">CSHL Single Cell Analysis - Bioinformatics</a> course materials - Uses Shalek 2013 and Macaulay 2016 datasets to teach machine learning to biologists</li>
<li>
<a href="http://jdblischak.github.io/singleCellSeq/analysis/">Gilad Lab Single Cell Data Exploration</a> - R-based exploration of single cell sequence data. Lots of experimentation. </li>
<li>
<a href="http://hms-dbmi.github.io/scw/">Harvard STEM Cell Institute Single Cell Workshop 2015</a> - workshop on common computational analysis techniques for scRNA-seq data from differential expression to subpopulation identification and network analysis. <a href="http://scholar.harvard.edu/jeanfan/classes/single-cell-workshop-2015">See course description for more information</a> </li>
<li>
<a href="http://hemberg-lab.github.io/scRNA.seq.course/index.html">Hemberg Lab scRNA-seq course materials</a> </li>
<li>
<a href="http://www.satijalab.org/seurat-intro.html">Using Seurat for unsupervised clustering and biomarker discovery</a> - 301 single cells across diverse tissues from (Pollen et al., Nature Biotechnology, 2014)</li>
<li>
<a href="http://www.satijalab.org/seurat-intro.html">Using Seurat for spatial inference in single-cell data</a> - 851 single cells from Zebrafish embryogenesis (Satija<em>, Farrell</em> et al., Nature Biotechnology, 2015)</li>
</ul>

<h2>
<a id="web-portals-and-apps" class="anchor" href="#web-portals-and-apps" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Web portals and apps</h2>

<ul>
<li>
<a href="https://portals.broadinstitute.org/single_cell">Neuro Single Cell Expression Portal at the Broad</a> - The Single-Cell RNA-Seq Portal for Brain Research was developed to facilitate sharing scientific results, and disseminate datasets resulting from the NIH's BRAIN initiative.</li>
</ul>

<h2>
<a id="journal-articles-of-general-interest" class="anchor" href="#journal-articles-of-general-interest" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Journal articles of general interest</h2>

<h3>
<a id="experimental-design" class="anchor" href="#experimental-design" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Experimental design</h3>

<ul>
<li><a href="http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0927-y">Design and computational analysis of single-cell RNA-sequencing experiments</a></li>
</ul>

<h3>
<a id="methods-comparisons" class="anchor" href="#methods-comparisons" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Methods comparisons</h3>

<ul>
<li>
<a href="http://biorxiv.org/content/early/2016/06/29/035758">Comparative analysis of single-cell RNA sequencing methods</a> - a comparison of wet lab protocols for scRNA sequencing.</li>
</ul>

<h2>
<a id="similar-lists-and-collections" class="anchor" href="#similar-lists-and-collections" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Similar lists and collections</h2>

<ul>
<li>
<a href="https://github.com/crazyhottommy/RNA-seq-analysis#single-cell-rna-seq">CrazyHotTommy's RNA-seq analysis list</a> - Very broad list that includes some single cell RNA-seq packages and papers.</li>
</ul>

<h2>
<a id="people" class="anchor" href="#people" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>People</h2>

<p>Gender bias at conferences is a well known problem (<a href="http://www.sciencemag.org/careers/2015/07/countering-gender-bias-conferences">http://www.sciencemag.org/careers/2015/07/countering-gender-bias-conferences</a>). Creating a list of potential speakers can help mitigate this bias and a community of people developing and maintaining helps to further diversify this list beyond smaller networks. </p>

<h3>
<a id="female" class="anchor" href="#female" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Female</h3>

<ul>
<li>
<a href="https://www.biostat.wisc.edu/%7Ekendzior/">Christina Kendziorski (University of Wisconsin‚ÄìMadison, USA)</a> </li>
<li><a href="http://www.stat.berkeley.edu/%7Esandrine/">Sandrine Dudoit (UC Berkeley, USA)</a></li>
<li><a href="http://bcb.dfci.harvard.edu/%7Ekeegan/">Keegan Korthauer (Dana Farber Cancer Institute, USA)</a></li>
<li><a href="http://www.stephaniehicks.com/">Stephanie Hicks (Dana Farber Cancer Institute, USA)</a></li>
<li><a href="http://www.c2b2.columbia.edu/danapeerlab/html/">Dana Pe'er (Columbia University, USA</a></li>
<li><a href="https://www.broadinstitute.org/scientific-community/science/core-faculty-labs/regev-lab/regev-lab-home">Aviv Regev (Broad Institute, USA)</a></li>
<li><a href="http://www.mrc-bsu.cam.ac.uk/people/in-alphabetical-order/t-to-z/catalina-vallejos-menesses/">Catalina Vallejos (MRC Biostatistics Unit and EMBL-EBI, UK)</a></li>
<li><a href="http://www.teichlab.org/">Sarah Teichmann (Wellcome Trust Sanger Institute, UK)</a></li>
<li><a href="http://cs.stanford.edu/people/emmap1/">Emma Pierson (Stanford University, USA)</a></li>
<li><a href="https://www.biostat.wisc.edu/%7Eningleng/">Ning Leng (Morgridge Institute for Research, USA)</a></li>
<li><a href="https://www.helmholtz-muenchen.de/icb/institute/staff/staff/ma/2453/-Haghverdi/index.html">Laleh Haghverdi (Institute of Computational Biology, Germany)</a></li>
</ul>

<h3>
<a id="male" class="anchor" href="#male" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Male</h3>

<ul>
<li><a href="https://www.fredhutch.org/en/labs/profiles/gottardo-raphael.html">Raphael Gottardo (Fred Hutchinson Cancer Research Center, USA)</a></li>
<li><a href="http://www.ebi.ac.uk/research/marioni">John Marioni (EBI, UK)</a></li>
<li><a href="http://www.ebi.ac.uk/research/stegle">Oliver Stegle (EBI, UK)</a></li>
<li><a href="https://sites.google.com/site/davismcc/">Davis McCarthy (EBI, UK)</a></li>
<li><a href="http://www.cruk.cam.ac.uk/users/aaron-lun">Aaron Lun (Cancer Research UK, UK)</a></li>
</ul>
        </section>

        <aside id="sidebar">
          <a href="https://github.com/seandavi/awesome-single-cell/zipball/master" class="button">
            <small>Download</small>
            .zip file
          </a>
          <a href="https://github.com/seandavi/awesome-single-cell/tarball/master" class="button">
            <small>Download</small>
            .tar.gz file
          </a>

          <p class="repo-owner"><a href="https://github.com/seandavi/awesome-single-cell"></a> is maintained by <a href="https://github.com/seandavi">seandavi</a>.</p>

          <p>This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the Architect theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.</p>
        </aside>
      </div>
    </div>

  
  </body>
</html>
