# STUFF FOR ME TO ADD


# Normalization

http://michelebusby.tumblr.com/post/130202229486/the-ks-test-looks-pretty-good-for-single-cell

Accounting for technical variation: http://www.nature.com/ncomms/2015/151022/ncomms9687/full/ncomms9687.html#supplementary-information

ZIFA: Zero-inflated factor analysis https://github.com/epierson9/ZIFA

http://biorxiv.org/content/early/2016/04/22/049734.full.pdf+html

https://www.dropbox.com/s/pno78mmlj0exv7s/NODES_0.0.0.9010.tar.gz?dl=0

scran: http://bioconductor.org/packages/devel/bioc/html/scran.html

Correct for expression heterogeneity: https://github.com/PMBio/scLVM

# Transcript counting

Modified version of Kallisto: https://github.com/govinda-kamath/clustering_on_transcript_compatibility_counts

DISCO: https://pbtech-vc.med.cornell.edu/git/mason-lab/disco/tree/master

# Clustering

Comparative analysis: http://biorxiv.org/content/early/2016/04/07/047613

destiny: diffusion maps for single-cell data http://bioconductor.org/packages/release/bioc/html/destiny.html

https://github.com/govinda-kamath/clustering_on_transcript_compatibility_counts

GiniClust https://github.com/lanjiangboston/GiniClust

pcaReduce: https://github.com/JustinaZ/pcaReduce

https://github.com/BatzoglouLabSU/SIMLR

# Differential Expression

Monocle cole-trapnell-lab.github.io/monocle-release/

scDD: https://github.com/kdkorthauer/scDD

ISOP: comparison of isoform pairs in single cells https://github.com/nghiavtr/ISOP

D3E: http://hemberg-lab.github.io/D3E/

BASiCS: https://github.com/catavallejos/BASiCS

Beta Poisson: https://github.com/nghiavtr/BPSC

# Time-series/ordering/lineage prediction

Monocle

Analysis of pseudotime uncertainty: http://biorxiv.org/content/biorxiv/early/2016/04/05/047365.full.pdf

ECLAIR: cell lineage prediction https://github.com/GGiecold/ECLAIR

Identification of ordering effects: https://github.com/lengning/OEFinder

Slicer: non-linear trajectories https://github.com/jw156605/SLICER

Wishbone: identification of bifurcations in developmental trajectories http://www.c2b2.columbia.edu/danapeerlab/html/cyt-download.html

SCOUP: https://github.com/hmatsu1226/SCOUP

Ouija: https://github.com/kieranrcampbell/ouija

# Pipelines

Seurat http://www.satijalab.org/seurat.html

SINCERA https://research.cchmc.org/pbge/sincera.html

MAST: https://github.com/RGLab/MAST

scde (differential expression + gene set over-dispersion): https://github.com/hms-dbmi/scde

BaSiCs: Bayesian analysis of single cell data: https://github.com/catavallejos/BASiCS

FastProject: https://github.com/YosefLab/FastProject/wiki

Citrus: http://chenmengjie.github.io/Citrus/

Tools from Teichmann lab (cellity, celloline, scrnatb): https://github.com/Teichlab/

SCell: https://github.com/diazlab/SCell
# Other

Ginko: analysis of CNVs in single-cell data: http://qb.cshl.edu/ginkgo/?q=/XWxZEerqqY477b9i4V8F

CNV calling: http://genome.cshlp.org/content/early/2016/01/15/gr.198937.115.full.pdf

Gene co-expression: http://journals.plos.org/ploscompbiol/article?id=10.1371%2Fjournal.pcbi.1004892

DNA SNV calling: https://bitbucket.org/hamimzafar/monovar

# Methylation

Prediction of missing information: https://github.com/cangermueller/deepcpg

# Reviews

Design and computational analysis of single-cell RNA-sequencing experiments: https://doi.org/10.1186/s13059-016-0927-y

Defining cell types and states with single-cell genomics: https://doi.org/10.1101/gr.190595.115

Scaling single-cell genomics from phenomenology to mechanism: https://doi.org/10.1038/nature21350
